src/Build/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -224,57 +222,67 @@ public void Consume(BuildEventArgs buildEvent, int sinkId)
         /// </summary>
         public void Consume(BuildEventArgs buildEvent)
         {
+
             switch (buildEvent)
             {
                 case BuildMessageEventArgs buildMessageEvent:
-                    RaiseMessageEvent(null, buildMessageEvent);
+                    RaiseEvent(buildMessageEvent, args => MessageRaised?.Invoke(null, args), RaiseAnyEvent);
                     break;
                 case TaskStartedEventArgs taskStartedEvent:
-                    RaiseTaskStartedEvent(null, taskStartedEvent);
+                    ArgsHandler<TaskStartedEventArgs> taskStartedFollowUp = args => RaiseEvent(args, args=> StatusEventRaised?.Invoke(null, args), RaiseAnyEvent);
```

Comment:
suggestion: RaiseEvent is called for StatusEventRaised and with RaiseAnyEvent as follow up multiple times. does it workk to define `ArgsHandler<BuildEventArgs> raiseStatusEvent = = args => RaiseEvent(args, args=>StatusEventRaised?.Invoke(null, args), RaiseAnyEvent);` ?

Reply:
That is lambda hiding some complexity there:
`ArgsHandler<BuildFinishedEventArgs> BuildFinishedFollowUp = args => RaiseEvent(args, args => StatusEventRaised?.Invoke(null, args), RaiseAnyEvent);`
vs
`ArgsHandler<BuildStartedEventArgs> BuildStartedFollowUp = args => RaiseEvent(args, args => StatusEventRaised?.Invoke(null, args), RaiseAnyEvent);`

I've tried some similar stuff, but it usually ends up with the compiler complaining about type mismatch. Maybe I'm just not familiar enough with generics though?

Reply:
in that case, I'm okay with merging as it is now.